Source/WebCore/rendering/RenderTableSection.cpp

```diff
@@ -874,9 +874,13 @@ std::optional<LayoutUnit> RenderTableSection::firstLineBaseline() const
     if (!m_grid.size())
         return std::optional<LayoutUnit>();
 
+    LayoutUnit tableMarginBefore;
+    if (RenderTable* parentTable = table())
+        tableMarginBefore = parentTable->style().marginBefore().value();
+
     LayoutUnit firstLineBaseline = m_grid[0].baseline;
     if (firstLineBaseline)
-        return firstLineBaseline + m_rowPos[0];
+        return firstLineBaseline + m_rowPos[0] + tableMarginBefore;
```

Comment:
Not sure how this is related to the last baseline?

Reply:
It is not directly related to the last baseline implementation but to a test case for it. There were 2 expectation files we were rendering incorrectly (css-flexbox/flexbox-align-self-baseline-horiz-001a-expected.xhtml and flexbox-align-self-baseline-horiz-001b-expected.xhtml) 

These were using the first baseline for table elements and we were not computing the correct first baseline. This caused us to fail the tests even though we were rendering the actual test file correctly. This was the incorrect way to do it and I moved the logic to RenderTable::baselinePosition. Now it is similar to other objects (e.g. RenderBlock::baselinePosition).